pkg/credentialprovider/config.go

```diff
@@ -30,6 +30,13 @@ import (
 	"github.com/golang/glog"
 )
 
+// DockerConfigJson represents ~/.docker/config.json file info
+// see https://github.com/docker/docker/pull/12009
+type DockerConfigJson struct {
+	Auths       DockerConfig      `json:"auths"`
+	HttpHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
Can you add a comment about what this is for?  (Based on the docs here: https://docs.docker.com/reference/commandline/cli/, it looks like a set of blindly attached headers) and an indication that we don't use or respect them (yet).

@erictune Are you willing to take the `HttpHeaders` support as a follow on item?  I think we'd gain a lot of benefit even without plumbing them through.